Core/MMKV_IO.cpp

```diff
@@ -790,6 +817,38 @@ MMKV::doAppendDataWithKey(const MMBuffer &data, const MMBuffer &keyData, bool is
     return make_pair(true, KeyValueHolder(originKeyLength, valueLength, offset));
 }
 
+KVHolderRet_t MMKV::overrideDataWithKey(const MMBuffer &data, MMKVKey_t key, bool isDataHolder) {
+    size_t old_actualSize = m_actualSize;
+    size_t old_position = m_output->getPosition();
+    // only one key in dict, do not append, just rewrite from beginning
+    m_actualSize = 0;
+    m_output->setPosition(0);
+
+    auto m_tmpDic = m_dic;
+    auto m_tmpDicCrypt = m_dicCrypt;
+    if (m_crypter) {
+        m_dicCrypt = new MMKVMapCrypt();
```

Comment:
This empty dummy dictionary works, but it looks ugly. Is there any way to avoid it?

Reply:
This is the simplest and most efficient solution I can think of at the moment: reuse present codes to ensure sufficient space、append、set.

There should be another solution, for example
1、rewrite this logic totally, but may do a lot of work which could not so valuable, and make code expand
2、let m_tmpDic = nullptr, but we must check null pointers everywhere

If you have any better idea, please let me know, then I will improve it.